src/serializer/GeneratorDAG.js

```diff
@@ -76,6 +76,22 @@ export class GeneratorDAG {
         this.createdObjects.set(createdObject, generator);
       }
 
+    if (parent instanceof Generator) {
+      // We'd like to ensure at this point that the path conditions
+      // of a nested generator imply the path conditions of its parent
+      // generator.
+      // Without a fancy solver, we simply rely on a very simple check
+      // whether the conditions of the parent generator are all
+      // present in the nested generator.
+      // TODO: This relies on path conditions not being simplified, which seems to be the case so far.
```

Comment:
Path conditions are in fact simplified. When you push a new path condition, all the existing path conditions are simplified using the new condition.